app/scripts/modules/core/src/pipeline/config/parameters/Parameters.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { every } from 'lodash';
+import { arrayMove, SortableContainer, SortableElement, SortEnd } from 'react-sortable-hoc';
+
+import { HelpField } from 'core/help';
+
+import { IParameter, IParameterProps, Parameter } from './Parameter';
+
+export interface IParametersState {
+  allParametersPinned: boolean;
+}
+
+export interface IParametersProps {
+  parameters: IParameter[];
+  pipelineName: string;
+  addParameter: () => void;
+  removeParameter: (index: number) => void;
+  updateParameter: (index: number, changes: { [key: string]: any }) => void;
+  updateAllParameters: (parameters: IParameter[]) => void;
+}
+
+export class Parameters extends React.Component<IParametersProps, IParametersState> {
+  constructor(props: IParametersProps) {
+    super(props);
+    this.state = {
+      allParametersPinned: false,
+    };
+  }
+
+  public componentDidMount() {
+    this.setPinAllParametersState();
+  }
+
+  private setPinAllParametersState = (): void => {
+    this.setState({ allParametersPinned: every(this.props.parameters || [], { pinned: true }) });
```

Comment:
```suggestion
    this.setState({ allParametersPinned: (this.props.parameters || []).every(p => p.pinned) });
```
this should work without bring lodash in

Reply:
👍